assets/js/data/checkout/constants.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { getSetting, EnteredAddress } from '@woocommerce/settings';
+
+export const STORE_KEY = 'wc/store/checkout';
+
+export enum STATUS {
+	// Checkout is in it's initialized state.
+	PRISTINE = 'pristine',
+	// When checkout state has changed but there is no activity happening.
+	IDLE = 'idle',
+	// After BEFORE_PROCESSING status emitters have finished successfully. Payment processing is started on this checkout status.
+	PROCESSING = 'processing',
+	// After the AFTER_PROCESSING event emitters have completed. This status triggers the checkout redirect.
+	COMPLETE = 'complete',
+	// This is the state before checkout processing begins after the checkout button has been pressed/submitted.
+	BEFORE_PROCESSING = 'before_processing',
+	// After server side checkout processing is completed this status is set
+	AFTER_PROCESSING = 'after_processing',
+}
+
+const preloadedCheckoutData = getSetting( 'checkoutData', {} ) as Record<
+	string,
+	unknown
```

Comment:
Yes but not always. It can return an empty object too, and I'm not sure if it can return part of the response object only. I've changed it to `Partial<CheckoutResponseSuccess>` to account for this. 👍